new release notes (#6916)
harness/developer-hub: release-notes/cloud-cost-management.md

:::
## May 2024
### Version 1.19.1

#### New features and enhancements

> - Asset Governance for AWS : Added new governance recommendations and cost computation support for AWS resources including `cache-cluster`, `S3`,`` Redshift`, and `Redshift-snapshot`. [CCM-17852]

Reviewer: ```suggestion
- Asset Governance for AWS : Added new governance recommendations and cost computation support for AWS resources including `cache-cluster`, `S3`, `Redshift`, and `Redshift-snapshot`. [CCM-17852]
```

- Editing Enforcements: Improved the enforcement editing process. Now, when viewing and editing an enforcement, it will no longer get enabled automatically if it was previously disabled. This ensures that the enforcement status remains consistent unless intentionally changed by the user. [CCM-18050]

> - Email Validation Enhancement: We've introduced better validation for email addresses in the recipients list for perspectives report. This enhancement ensures that only correctly formatted email addresses are accepted in the recipients list, enhancing data integrity and security. [CCM-17850]

Reviewer: ```suggestion
- Email Validation Enhancement: We've introduced better validation for email addresses in the recipients list for perspectives report. This enhancement ensures that only correctly formatted email addresses are accepted in the recipient's list, enhancing data integrity and security. [CCM-17850, ZD-63324]
```

Author: hey @brian-f-harness the list is actually the recipients list

#### Fixed issues

> - Azure perspectives Previously, only fields in "group by" with available data were displayed in Azure perspectives, leading to incomplete views and unnecessary errors. With this update, all relevant fields, including those without data, will now be visible. This ensures a comprehensive perspective of the data, eliminating unnecessary errors arising from missing data at the source level. [CCM-17573]

Reviewer: ```suggestion
- Azure perspectives Previously, only fields in "group by" with available data were displayed in Azure perspectives, leading to incomplete views and unnecessary errors. With this update, all relevant fields, including those without data, will be visible. This ensures a comprehensive data perspective, eliminating unnecessary errors from missing data at the source level. [CCM-17573, ZD-62691]
```

- Email Validation Enhancement: We've introduced better validation for email addresses in the recipients list for perspectives report. This enhancement ensures that only correctly formatted email addresses are accepted in the recipients list, enhancing data integrity and security. [CCM-17850, ZD-63324]
- Azure Governance Subscription selection: In the Azure Rule window's Subscription drop-down menu, previously, only the Subscription ID was displayed. Now, both the Subscription Name and ID are shown for better clarity and ease of selection. [CCM-17650]
- Node New K8s Labelling: We have added support for the new K8s labels starting from K8s v1.17 for Instance Type, Operating System, Region, and Zone respectively. We use these labels to get the public pricing data for a given cloud provider. [CCM-17979]

| Old Label | New Label |
|------------|--------------|
| beta.kubernetes.io/instance-type | node.kubernetes.io/instance-type |
| beta.kubernetes.io/os | kubernetes.io/os |
| failure-domain.beta.kubernetes.io/region | topology.kubernetes.io/region |
| failure-domain.beta.kubernetes.io/zone | topology.kubernetes.io/zone |

> #### Fixed issues

Reviewer 2: Fixed an issue in Azure perspectives which displayed only the fields in the "group by" filter with the available data. This led to errors. Now, all the relevant fields, including those without data, will be visible. This eliminates unnecessary errors arising from missing data at the source level. [CCM-17573, ZD-62691]

#### New features and enhancements

- Asset Governance for AWS : Added new governance recommendations and cost computation support for AWS resources including `cache-cluster`, `S3`,`Redshift`, and `Redshift-snapshot`. [CCM-17852]

> - Asset Governance for GCP : Added new governance recommendations and cost computation support for GCP resources including `redis`, `gke-cluster`,` bq-dataset`, `function`, `bucket`, `dataflow-job`, `loadbalancer-address`, `cloud-run-service` and also, added cost computation for `gcp.loadbalancer-forwarding-rule`. [CCM-17852]

Reviewer: ```suggestion
- Asset Governance for GCP : Added new governance recommendations and cost computation support for GCP resources including `redis`, `gke-cluster`, `bq-dataset`, `function`, `bucket`, `dataflow-job`, `loadbalancer-address`, `cloud-run-service` and also, added cost computation for `gcp.loadbalancer-forwarding-rule`. [CCM-17852]
```

- Editing Enforcements: Improved the enforcement editing process. Now, when viewing and editing an enforcement, it will no longer get enabled automatically if it was previously disabled. This ensures that the enforcement status remains consistent unless intentionally changed by the user. [CCM-18050]
- Email Validation Enhancement: We've introduced better validation for email addresses in the recipients list for perspectives report. This enhancement ensures that only correctly formatted email addresses are accepted in the recipients list, enhancing data integrity and security. [CCM-17850, ZD-63324]
- Azure Governance Subscription selection: In the Azure Rule window's Subscription drop-down menu, previously, only the Subscription ID was displayed. Now, both the Subscription Name and ID are shown for better clarity and ease of selection. [CCM-17650]

> - Node New K8s Labelling: We have added support for the new K8s labels starting from K8s v1.17 for Instance Type, Operating System, Region and Zone respectively. We use these labels to get the public pricing data for a given cloud provider. [CCM-17979]

Reviewer: ```suggestion
- Node New K8s Labelling: We have added support for the new K8s labels starting from K8s v1.17 for Instance Type, Operating System, Region, and Zone respectively. We use these labels to get the public pricing data for a given cloud provider. [CCM-17979]
```
